framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -3463,17 +3461,13 @@ private AnnotatedDeclaredType getFunctionalInterfaceType(
         }
     }
 
-    private void assertFunctionalInterface(
-            com.sun.tools.javac.code.Types javacTypes,
-            Type type,
-            Tree contextTree,
-            Tree lambdaTree) {
+    private void assertFunctionalInterface(Type type, Tree contextTree, Tree lambdaTree) {
```

Comment:
How about making the first argument a `TypeMirror` and casting here, to safe all the casts in the invocations?